File: lib/domain/field_cultivation/errors/field_cultivation_sync_duplicate_allocation_error.rb
# frozen_string_literal: true

module Domain
  module FieldCultivation
    module Errors
      class FieldCultivationSyncDuplicateAllocationError < StandardError
        attr_reader :duplicate_ids

        def initialize(duplicate_ids:)
          @duplicate_ids = duplicate_ids
          super("duplicate allocation ids: #{duplicate_ids.join(', ')}")
        end
      end
    end
  end
end

File: crates/agrr-domain/src/field_cultivation/errors/field_cultivation_sync_duplicate_allocation_error.rs
use thiserror::Error;

/// Ruby: `Domain::FieldCultivation::Errors::FieldCultivationSyncDuplicateAllocationError`
#[derive(Debug, Clone, PartialEq, Eq, Error)]
#[error("duplicate allocation ids: {duplicate_ids:?}")]
pub struct FieldCultivationSyncDuplicateAllocationError {
    pub duplicate_ids: Vec<String>,
}

impl FieldCultivationSyncDuplicateAllocationError {
    pub fn new(duplicate_ids: Vec<String>) -> Self {
        Self { duplicate_ids }
    }
}
